builtin/credential/radius/path_config.go

```diff
@@ -0,0 +1,224 @@
+package radius
+
+import (
+	"strings"
+
+	"github.com/fatih/structs"
+	"github.com/hashicorp/vault/logical"
+	"github.com/hashicorp/vault/logical/framework"
+)
+
+func pathConfig(b *backend) *framework.Path {
+	return &framework.Path{
+		Pattern: `config`,
+		Fields: map[string]*framework.FieldSchema{
+			"host": &framework.FieldSchema{
+				Type:        framework.TypeString,
+				Default:     "127.0.0.1",
+				Description: "radius host to connect to (default: 127.0.0.1)",
+			},
+
+			"port": &framework.FieldSchema{
+				Type:        framework.TypeInt,
+				Default:     1812,
+				Description: "radius port (default: 1812)",
+			},
+			"secret": &framework.FieldSchema{
+				Type:        framework.TypeString,
+				Description: "secret shared with the radius server",
+			},
+			"allow_unknown_users": &framework.FieldSchema{
+				Type:        framework.TypeBool,
+				Default:     false,
+				Description: "enable granting default policy upon successful RADIUS authentication (default: false)",
+			},
+			"reauth_on_renew": &framework.FieldSchema{
+				Type:        framework.TypeBool,
+				Default:     false,
+				Description: "attempt reauthentication with backend before granting token renewal (default: false)",
+			},
+			"dial_timeout": &framework.FieldSchema{
+				Type:        framework.TypeDurationSecond,
+				Default:     10,
+				Description: "number of seconds before connect timeouts (default: 10)",
+			},
+			"read_timeout": &framework.FieldSchema{
+				Type:        framework.TypeDurationSecond,
+				Default:     10,
+				Description: "number of seconds before response timeouts (default: 10)",
+			},
+			"nas_port": &framework.FieldSchema{
+				Type:        framework.TypeInt,
+				Default:     10,
+				Description: "RADIUS NAS port field (default: 10)",
+			},
+		},
+
+		Callbacks: map[logical.Operation]framework.OperationFunc{
+			logical.ReadOperation:   b.pathConfigRead,
+			logical.UpdateOperation: b.pathConfigWrite,
+		},
+
+		HelpSynopsis:    pathConfigHelpSyn,
+		HelpDescription: pathConfigHelpDesc,
+	}
+}
+
+/*
+ * Construct ConfigEntry struct using stored configuration.
+ */
+func (b *backend) Config(req *logical.Request) (*ConfigEntry, error) {
+	// Schema for ConfigEntry
+	fd, err := b.getConfigFieldData()
+	if err != nil {
+		return nil, err
+	}
+
+	// Create a new ConfigEntry, filling in defaults where appropriate
+	result, err := b.newConfigEntry(fd)
+	if err != nil {
+		return nil, err
+	}
+
+	storedConfig, err := req.Storage.Get("config")
+	if err != nil {
+		return nil, err
+	}
+
+	if storedConfig == nil {
+		// No user overrides, return default configuration
+		return result, nil
```

Comment:
Added checks that will throw an error on auth when missing required config params are missing:

example:
```
(default)➜  ~ vault auth -method=userpass username=test
Password (will be hidden):
Error making API request.

URL: PUT http://127.0.0.1:8200/v1/auth/userpass/login/test
Code: 400. Errors:

* radius backend not configured
```